impeller/geometry/matrix.h

```diff
@@ -313,6 +313,10 @@ struct Matrix {
             m[9] == 0 && m[10] == 1 && m[11] == 0 && m[14] == 0 && m[15] == 1);
   }
 
+  constexpr bool HasPerspective() const {
+    return m[3] != 0 || m[7] != 0 || m[11] != 0 || m[15] != 1;
```

Comment:
That should be safe. We may not care about 11 for the case at hand since it is the Z component of W and Z=0, but it can't hurt to bail on that for simplicity/robustness sake.